SpatialGDK/Source/SpatialGDK/Private/Utils/RPCContainer.cpp

```diff
@@ -4,28 +4,155 @@
 
 #include "Schema/UnrealObjectRef.h"
 
+DEFINE_LOG_CATEGORY(LogRPCContainer);
+
 using namespace SpatialGDK;
 
-FPendingRPCParams::FPendingRPCParams(const FUnrealObjectRef& InTargetObjectRef, SpatialGDK::RPCPayload&& InPayload, int InReliableRPCIndex /* = 0 */)
-	: ReliableRPCIndex(InReliableRPCIndex)
-	, ObjectRef(InTargetObjectRef)
+namespace
+{
+	FString ERPCResultToString(ERPCResult Error)
+	{
+		switch (Error)
+		{
+		case ERPCResult::Success:
+			return TEXT("");
+
+		case ERPCResult::NoProcessingFunctionBound:
+			return TEXT("No Processing Function Bound");
+
+		case ERPCResult::UnresolvedTargetObject:
+			return TEXT("Unresolved Target Object");
+
+		case ERPCResult::MissingFunctionInfo:
+			return TEXT("Missing UFunction info");
+
+		case ERPCResult::UnresolvedParameters:
+			return TEXT("Unresolved Parameters");
+
+		case ERPCResult::NoActorChannel:
+			return TEXT("No Actor Channel");
+
+		case ERPCResult::SpatialActorChannelNotListening:
+			return TEXT("Spatial Actor Channel Not Listening");
+
+		case ERPCResult::NoNetConnection:
+			return TEXT("No Net Connection");
+
+		case ERPCResult::NoAuthority:
+			return TEXT("No Authority");
+
+		case ERPCResult::InvalidRPCType:
+			return TEXT("Invalid RPC Type");
+
+		case ERPCResult::NoOwningController:
+			return TEXT("No Owning Controller");
+
+		case ERPCResult::NoControllerChannel:
+			return TEXT("No Controller Channel");
+
+		case ERPCResult::UnresolvedController:
+			return TEXT("Unresolved Controller");
+
+		case ERPCResult::ControllerChannelNotListening:
+			return TEXT("Controller Channel Not Listening");
+
+		default:
+			return TEXT("Unknown");
```

Comment:
would a checknoentry here be appropriate?

Reply:
This function is used for error logging only, so I thought it's a case when it's fine to not have a checknoentry. But I'm fine with adding it.